### PR TITLE
jewel: list-inconsistent-obj and list-inconsistent-snapset improvements #9467

### PR DESCRIPTION
**Pull request by chardan**
http://tracker.ceph.com/issues/15767

### CODE REVIEW

**Comment by dzafman** [comment]
Jenkins failed, so I assume that a conflict wasn't resolved correctly.

**Comment by chardan** [comment]
dzafman: Definitely possible-- I have to say that we happily sent it off to Jenkins to build rather than trying it locally first this time.

File: src/tools/rados/rados.cc [hunk 1]
```diff
@@ -1338,18 +1350,40 @@ static void dump_inconsistent(const inconsistent_snapset_t& inc,
 
   if (inc.clone_missing()) {
```
Comment (tchaikov) [inline src/tools/rados/rados.cc]: should remove these two lines.

File: src/tools/rados/rados.cc [hunk 2]
```diff
@@ -1338,18 +1350,40 @@ static void dump_inconsistent(const inconsistent_snapset_t& inc,
...
+  if (inc.size_mismatch())
+    f.dump_bool("size_mismatch", inc.size_mismatch());
+
+  if (inc.extra_clones()) {
+    f.dump_bool("extra_clones", inc.extra_clones());
+    f.open_array_section("extra clones");
     for (auto snap : inc.clones) {
       f.dump_unsigned("snap", snap);
     }
     f.close_section();
```
Comment (tchaikov) [inline src/tools/rados/rados.cc]: and add a "`}`" here.

**Comment by chardan** [comment]
Hope I got this right-- if not, we'll sort it out. Thanks for your patience!

**Comment by dzafman** [comment]
Try re-creating this branch with a cherry pick -x of the commits in this order:

cea7cf56ee3b9593c935d0c74ce6e4b28b14c2da df4bc16c4d49489f7398eb1ecc7b9aef93258414 de4681b80a5d24e72bbc7ac6dfee4775987fd834 4fc4326b36aa00092043514746aa39aedad06b0f fdca113fc24adbb4f60bfdc55fbbd5044b711b5d a551b6bd5c4eba11c170afe63994258e7598c3d2 db517ba9176f1ee829453b016b1bd43c6054a555

I don't get a conflict.  I don't know why you have 8 commits in this pull request.

**Comment by dzafman** [comment]
I'll create a new pull request.

**Pull request closed without merging**